lib/v2/screens/sign_up/claim_invite_screen.dart

```diff
@@ -0,0 +1,190 @@
+            Text(
+              'Or enter by yourself below',
+              style: Theme.of(context).accentTextTheme.subtitle2,
+            ),
+            const SizedBox(height: 16),
+            TextFormField(
+              controller: _keyController,
+              style: Theme.of(context).accentTextTheme.bodyText1,
+              onChanged: _onInviteCodeChanged,
+              decoration: InputDecoration(
+                hintText: 'Invite code (5 words)'.i18n,
+                hintStyle: Theme.of(context).accentTextTheme.bodyText1,
+                errorText: _currentState.pageState == PageState.failure
+                    ? _currentState.errorMessage
+                    : null,
+                suffixIcon: _inviteCodeSuffixIcon,
+                enabledBorder: const OutlineInputBorder(
+                  borderRadius: BorderRadius.all(Radius.circular(8)),
+                  borderSide: BorderSide(
+                    width: 1,
+                    color: Colors.black,
+                  ),
+                ),
+                focusedBorder: const OutlineInputBorder(
+                  borderRadius: BorderRadius.all(Radius.circular(8)),
+                  borderSide: BorderSide(
+                    width: 1,
+                    color: Colors.black,
+                  ),
+                ),
+                errorBorder: const OutlineInputBorder(
+                  borderRadius: BorderRadius.all(Radius.circular(8)),
+                  borderSide: BorderSide(
+                    width: 1,
+                    color: AppColors.red,
+                  ),
+                ),
+              ),
+            ),
+            const SizedBox(height: 50),
+            FlatButtonLong(
+              title: "Claim code".i18n,
+              onPressed: _onClaimPressed(context),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Widget get _inviteCodeSuffixIcon {
+    return _currentState.pageState == PageState.success &&
+            _currentState.inviteModel != null
+        ? const Icon(
+            Icons.check_circle,
+            color: AppColors.green,
+          )
+        : IconButton(
+            icon: const Icon(
+              Icons.paste,
+              color: AppColors.grey,
+            ),
+            onPressed: () async {
+              var clipboardData = await Clipboard.getData('text/plain');
+              var clipboardText = clipboardData?.text ?? '';
+              setState(() {
+                _keyController.text = clipboardText;
+              });
+            },
+          );
+  }
+
+  void _onInviteCodeChanged(String value) {
+    if (value.isNotEmpty) {
+      _debouncer.run(() {
+        _signupBloc.add(ValidateInviteCode(inviteCode: value));
+      });
+    }
+  }
+
+  void _onQRScanned(String scannedLink) {
+    _signupBloc.add(UnpackScannedLink(scannedLink));
+  }
+
+  VoidCallback? _onClaimPressed(BuildContext context) {
+    final bool isValid = _currentState.pageState == PageState.success &&
```

Comment:
sholdnt this be an event that fires into the Bloc? 
Then the bloc can handle this logic and fire a command back to the UI

All this logic can be moved away from the UI

Reply:
I think it's more of a preference but on my other projects and bloc's documentation, navigation is handled in the UI instead of going through a block. Anything that changes the state should go through bloc though.

Reply:
I think if its simple, like  only `NavigationService.of(context).navigateTo(Routes.displayName);`
Then sure, just go ahead an navigate, 

 But when you need to start having logic then i like to move it to the bloc away from the UI. 
 
 I tell the guys to follow "No logic in the UI" as much as possible. 
 
 Also, lets say that you apart from navigating, you need to send an analytic event to a server? The bloc should handle this, not the UI.